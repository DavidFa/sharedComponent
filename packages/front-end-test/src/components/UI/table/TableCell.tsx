import React from 'react'
import styled from 'styled-components';

const Td = styled.td`
width:300px;
font-family: Roboto, Helvetica, Arial, sans-serif;
font-weight: 500;
font-size: 0.875rem;
line-height: 1.5rem;
letter-spacing: 0em;
display: table-cell;
vertical-align: inherit;
`;

const CheckBox = styled.input``

type TableCellType = {
    selectable?: boolean;
    postId?: number;
    onSelectedHandler?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const TableCell: React.FC<TableCellType> = ({ selectable = false, postId, onSelectedHandler, children }) => {

    return (
        <Td>{selectable ? <CheckBox type="checkbox" value={postId} onChange={onSelectedHandler} /> : children}</Td>
    )
}

export default TableCell;
