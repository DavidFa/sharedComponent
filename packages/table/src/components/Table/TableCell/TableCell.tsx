import React from "react";
import styled from "styled-components";

type TableCellType = {
    width?: number;
    color?: string;
    background?: string;
    borderBottom?: boolean;
    colSpan?: number;
    textAlign?: string;
}

interface CellAttrProps {
    readonly colSpan: number;
}

interface CellProps {
    readonly width?: number;
    readonly color?: string;
    readonly background?: string;
    readonly borderBottom?: boolean;
    readonly textAlign?: string;
}

const Cell = styled.td.attrs<CellAttrProps>(props => ({
    colSpan: props.colSpan
})) <CellProps>`
    width:${props => props.width ? props.width : "100"}px;
    font-family: Roboto, Helvetica, Arial, sans-serif;
    font-weight: 500;
    font-size: 0.875rem;
    line-height: 1.5rem;
    letter-spacing: 0em;
    display: table-cell;
    vertical-align: inherit;
    border-bottom: ${props => props.borderBottom ? "1px solid rgb(224, 224, 224)" : ""};
    text-align:${props => props.textAlign ? props.textAlign : "right"};
    padding: 12px 24px;
    background: ${props => props.background ? props.background : "#fff"};
    color: ${props => props.color ? props.color : "rgba(0, 0, 0, 0.87)"};
`;


const TableCell: React.FC<TableCellType> = (props) => {

    return <Cell {...props}>{props.children}</Cell>
}

export default TableCell;