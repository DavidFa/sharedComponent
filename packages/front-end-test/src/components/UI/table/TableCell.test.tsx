import TableCell from "./TableCell";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';

describe("TableCell", () => {
  describe("Snapshot Test", () => {
    it('should render with checkbox', () => {
      const component = renderer.create(
        <TableCell selectable={true} postId={1}>
          Table Cell Data
        </TableCell>
      )
      expect(component.toJSON()).toMatchSnapshot();
    })

    it('should render without checkbox', () => {
      const component = renderer.create(
        <TableCell postId={1}>
          Table Cell Data
        </TableCell>
      )
      expect(component.toJSON()).toMatchSnapshot();
    })
  });



  describe("Dom Test", () => {
    const table = document.createElement("table");
    const tbody = document.createElement("tbody");
    const tr = document.createElement("tr");
    const tableContainer = document.body.appendChild(table).appendChild(tbody).appendChild(tr);

    it("should render with checkbox", () => {
      const { container } = render(
        <TableCell selectable={true} postId={1}>
          Table Cell Data
        </TableCell>,
        {
          container: tableContainer
        }
      );

      const td = screen.queryByRole("cell");
      expect(td).toBeInTheDocument();

      expect(container.firstChild).toMatchInlineSnapshot(`
          <td
            class="sc-bdfBQB bjKVnb"
            role="cell"
          >
            <input
              class="sc-gsTEea"
              type="checkbox"
              value="1"
            />
          </td>
        `);
    });

    it("should render with checkbox", () => {
      const { container } = render(
        <TableCell selectable={true} postId={1}>
          Table Cell Data
        </TableCell>,
        {
          container: tableContainer
        }
      );
      
      const checkbox = screen.queryByRole("checkbox");
      expect(checkbox).toBeInTheDocument();

      expect(container.firstChild).toMatchInlineSnapshot(`
        <td
          class="sc-bdfBQB bjKVnb"
          role="cell"
        >
          <input
            class="sc-gsTEea"
            type="checkbox"
            value="1"
          />
        </td>
      `);
    });

    it("should render with checkbox", () => {
      const { container } = render(
        <TableCell postId={1}>Table Cell Data</TableCell>,
        {
          container: tableContainer
        }
      );

      const checkbox = screen.queryByRole("checkbox");
      expect(checkbox).not.toBeInTheDocument();

      expect(container.firstChild).toMatchInlineSnapshot(`
        <td
          class="sc-bdfBQB bjKVnb"
          role="cell"
        >
          Table Cell Data
        </td>
      `);
    });
  });
});
