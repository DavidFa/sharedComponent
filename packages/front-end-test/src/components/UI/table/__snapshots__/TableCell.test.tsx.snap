// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell Snapshot Test should render with checkbox 1`] = `
<td
  className="sc-bdfBQB bjKVnb"
  role="cell"
>
  <input
    className="sc-gsTEea"
    type="checkbox"
    value={1}
  />
</td>
`;

exports[`TableCell Snapshot Test should render without checkbox 1`] = `
<td
  className="sc-bdfBQB bjKVnb"
  role="cell"
>
  Table Cell Data
</td>
`;
